src/repluggableAppDebug/repluggableAppDebug.ts

```diff
@@ -40,15 +40,39 @@ export function setupDebugInfo({
     shellInstallers,
     performance: { options, trace, memoizedArr }
 }: SetupDebugInfoParams) {
+    let apisAccessObject: Record<string, any> | undefined
+    let apisAccessObjectSize: number
+
     const utils = {
-        apis: () => {
+        getApis: () => {
             return Array.from(readyAPIs).map((apiKey: AnySlotKey) => {
                 return {
                     key: apiKey,
                     impl: () => getAPI(apiKey)
                 }
             })
         },
+        get apis() {
+            // We need to use cache object otherwise chrome devtools doesn't enable autocomplete
+            if (apisAccessObject && apisAccessObjectSize !== readyAPIs.size) {
+                apisAccessObject = undefined
+            }
+            if (!apisAccessObject) {
```

Comment:
Extracting creation of a new `apisAccessObjectSize` object into a separate function would help avoiding multiple mutations of the variable.

I'd also extract the staleness check `apisAccessObjectSize !== readyAPIs.size` into a function, as it would give that expression a descriptive name.

BTW, the staleness check should be improved. Entry points can come and go, and together with them the contributed APIs. If one API goes and another one is added, the size check won't detect that.

I'd suggest adding a "host version number" to AppHost implementation. Every time an entry point is added or removed, the "host version number" is incremented by the AppHost. Comparing to last version number would make a hermetic check.

Reply:
Added readyAPIsVersion